### PR TITLE
Add ErrOutOfBoundsValue for value boundary validation

### PR DESCRIPTION
**Pull request by k4leung4**

### CODE REVIEW

File: apis/field_error.go [hunk 1]
```diff
@@ -335,3 +335,12 @@ func ErrInvalidKeyName(value, fieldPath string, details ...string) *FieldError {
 		Details: strings.Join(details, ", "),
 	}
 }
+
+// ErrOutOFBoundsValue constructs a FieldError for a field that has received an
+// out of bound value.
+func ErrOutOfBoundsValue(value, lower, upper float64, fieldPath string) *FieldError {
```
Comment (jonjohnsonjr) [inline apis/field_error.go]: I think this name is a bit too general for the signature being specifically float64.

Comment (k4leung4) [reply apis/field_error.go]: change it to take string.
ptal

File: apis/field_error.go [hunk 2]
```diff
@@ -335,3 +335,12 @@ func ErrInvalidKeyName(value, fieldPath string, details ...string) *FieldError {
 		Details: strings.Join(details, ", "),
 	}
 }
+
+// ErrOutOFBoundsValue constructs a FieldError for a field that has received an
+// out of bound value.
+func ErrOutOfBoundsValue(value, lower, upper, fieldPath string) *FieldError {
+	return &FieldError{
+		Message: fmt.Sprintf("expected %q <= %q <= %q", lower, value, upper),
```
Comment (mattmoor) [inline apis/field_error.go]: I'd drop the quotes, it makes it look funny.

Comment (k4leung4) [reply apis/field_error.go]: removed.

**Pull request merged**